### PR TITLE
Add `ignore_any_from_error` feature

### PR DESCRIPTION
**Pull request by KotlinIsland**
image

- closes: #272

### CODE REVIEW

File: mypy/options.py [hunk 1]
```diff
@@ -121,6 +121,7 @@ def __init__(self) -> None:
         self.auto_baseline = True
         self.default_return = False
         self.targets: List[str] = []
+        self.ignore_any_from_error = True
```
Comment (DetachHead) [inline mypy/options.py:124]: why is it defaulted to `True` here but `False` here https://github.com/KotlinIsland/basedmypy/pull/281/files#diff-207f26497af1585033935dce5b1b9e3a12f9b98f28a8a8d7bf7bbb1ca88375aeR549

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## [Unreleased]
 ### Added
+- `ignore_any_from_errors` option to suppress `no-any-expr` messages from other errors
```
Comment (DetachHead) [inline CHANGELOG.md:5]: i wouldf suggest adding some more documentation to clarify what this means

**Comment by DetachHead** [comment]
how does it work if the `Any` errors from a baselined error are suppressed? do they come back? (i think they probably should)

File: mypy/options.py (hunk 1 above)
Comment (KotlinIsland) [reply mypy/options.py:124]: `False` means `True` in the trippple negative cli framework.

File: CHANGELOG.md (hunk 2 above)
Comment (KotlinIsland) [reply CHANGELOG.md:5]: I'm gonna update the docs when it's merged.

**Comment by KotlinIsland** [comment]
baselined or type ignored will produce the same behavior
```py
a = AMONGUS  # type: ignore[name-defined]
c = a[0]
reveal_type(c)
```
```
> mypy test.py
test.py:3:13: note: Revealed type is "Any"
Success: no issues found in 1 source file
```

Do you think that if you ignore the error it should show the other errors?

File: mypy/options.py (hunk 1 above)
Comment (DetachHead) [reply mypy/options.py:124]: epic

**Comment by DetachHead** [comment]
idk, it just sounds like the same issue as https://github.com/KotlinIsland/basedmypy/issues/114#issuecomment-1010691007

**Comment by KotlinIsland** [comment]
In this example:
```py
a = AMONGUS  # type: ignore[name-defined]
c = a[0]
reveal_type(c)
```
Ideally you would get a second error saying that you need an explicit type.

File: test-data/unit/check-based-ignore-any-from-error.test [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+             # E:5: Expression has type "Any"  [no-any-expr]
+b = a + 1  # E:5: Expression has type "Any"  [no-any-expr]
+c = b()  # E:5: Expression has type "Any"  [no-any-expr]
+d = [c]  # E:5: Expression type contains "Any" (has type "List[Any]")  [no-any-expr] \
+         # E:6: Expression has type "Any"  [no-any-expr]
+e = d[0].d  # E:5: Expression type contains "Any" (has type "List[Any]")  [no-any-expr] \
+            # E:5: Expression has type "Any"  [no-any-expr]
+e + 1  # E:1: Expression has type "Any"  [no-any-expr]
+f: Any  # E:1: Explicit "Any" is not allowed  [no-any-explicit]
+g = f + 1  # E:5: Expression has type "Any"  [no-any-expr]
```
Comment (DetachHead) [inline test-data/unit/check-based-ignore-any-from-error.test:25]: didnt u change it so that line numbers are no longer in errors in the tests?

**Pull request merged**